### ISSUE TITLE
Vector cannot be cast to tensor

### ISSUE THREAD

**Issue by lstrsrmn** [vehicle-lang/vehicle#1171]
Currently, Vector cannot be cast to a tensor of one dimension for (at least) network definitions. This means that a spec like this in invalid:
```haskell
@network
f : Tensor Real [10] -> Vector Real 2
```
But this is valid
```haskell
f : Tensor Real [10] -> Tensor Real [2]
```

These are the same types functionally, so it would be nice if both versions were accepted

**Comment by MatthewDaggitt** [comment]
This is by design 😄  Vectors can contain any element type are simply sized lists. They are not compilable in general to tensors and therefore PyTorch/Tensorflow code. Is there a reason you want to use a `Vector` as an output of a neural network?

**Comment by lstrsrmn** [comment]
Not specifically. I found this because I needed extraction of vectors to the loss backend which isnt implemented (for the fashion mnist spec) and wanted to see if tensors could get around it, but it makes sense to keep them distinct.
